tests/dragonfly/conftest.py

```diff
@@ -327,39 +354,41 @@ def with_ca_tls_client_args(with_tls_client_args, with_tls_ca_cert_args):
     return args
 
 
-def copy_failed_logs_and_clean_tmp_folder(report):
-    return  # TODO: to fix it first and then enable it.
+def copy_failed_logs_and_clean_tmp_folder(log_dir, report):
     failed_path = "/tmp/failed"
-    path_exists = os.path.exists(failed_path)
-    if not path_exists:
-        os.makedirs(failed_path)
-
-    if os.path.isfile("/tmp/last_test_log_files.txt"):
-        last_log_file = open("/tmp/last_test_log_files.txt", "r")
-        files = last_log_file.readlines()
-        logging.error(f"Test failed {report.nodeid} with logs: ")
-        for file in files:
-            # copy to failed folder
-            file = file.rstrip("\n")
-            logging.error(f"🪵🪵🪵🪵🪵🪵 {file} 🪵🪵🪵🪵🪵🪵")
-            shutil.copy(file, failed_path)
-
-
-def pytest_exception_interact(node, call, report):
-    if report.failed:
-        copy_failed_logs_and_clean_tmp_folder(report)
-
-
-@pytest.fixture(autouse=True)
-def run_before_and_after_test():
-    # Setup: logic before any of the test starts
-    # Empty the log on each run
-    last_log_file = open("/tmp/last_test_log_files.txt", "w")
-    last_log_file.close()
-
-    yield  # this is where the testing happens
-
-    # Teardown
+    if os.path.exists(failed_path):
+        shutil.rmtree(failed_path)
+
+    os.makedirs(failed_path)
+
+    logging.error(f"Test failed {report.nodeid} with logs: ")
+    cwd = os.getcwd()
+    onlyfiles = [
+        os.path.join(log_dir, f)
+        for f in os.listdir(log_dir)
+        if os.path.isfile(os.path.join(log_dir, f))
```

Comment:
:exploding_head: Why not just open `/tmp/last_log_dir` and print only the log name ? (that way we can get rid of the folder etc)

Reply:
As I understand we can have several failed tests. In this case we need copy directories too to have ability distinguish log files for different tests

Reply:
We don;t need that. When a test fails we print the log names and the test parameters so we can always find them on the end folder. We usually don't have more than 1 test failing and even for more, it's almost instant to cross reference

Reply:
this approach is clearer, you just open folder and name of the folder shows you what this test was and which parameters it uses

Reply:
+1 if we can keep the names short